spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/CredentialsProviderSelfManagedTest.java

```diff
@@ -22,57 +22,86 @@
 import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
 import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
 import static com.github.tomakehurst.wiremock.client.WireMock.verify;
-import static org.assertj.core.api.Assertions.*;
 import static org.assertj.core.api.Assertions.assertThat;
 
 import com.auth0.jwt.JWT;
 import com.auth0.jwt.algorithms.Algorithm;
 import com.github.tomakehurst.wiremock.junit5.WireMockTest;
-import io.camunda.zeebe.spring.client.configuration.AuthenticationConfiguration;
-import io.camunda.zeebe.spring.common.auth.Authentication;
-import io.camunda.zeebe.spring.common.auth.Product;
-import io.camunda.zeebe.spring.common.auth.selfmanaged.SelfManagedAuthentication;
+import io.camunda.zeebe.client.CredentialsProvider;
+import io.camunda.zeebe.client.impl.oauth.OAuthCredentialsProvider;
+import io.camunda.zeebe.spring.client.configuration.JsonMapperConfiguration;
+import io.camunda.zeebe.spring.client.configuration.ZeebeClientConfigurationImpl;
+import io.camunda.zeebe.spring.client.jobhandling.ZeebeClientExecutorService;
+import io.camunda.zeebe.spring.client.properties.CamundaClientProperties;
+import io.camunda.zeebe.spring.client.properties.ZeebeClientConfigurationProperties;
+import java.io.IOException;
+import java.nio.file.Paths;
 import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
 import wiremock.com.fasterxml.jackson.databind.node.JsonNodeFactory;
 
 @SpringBootTest(
-    classes = {AuthenticationConfiguration.class},
+    classes = {JsonMapperConfiguration.class, ZeebeClientConfigurationImpl.class},
     properties = {
       "camunda.client.mode=oidc",
       "camunda.client.auth.client-id=my-client-id",
       "camunda.client.auth.client-secret=my-client-secret",
       "camunda.client.auth.issuer=http://localhost:14682/auth-server"
     })
+@EnableConfigurationProperties({
+  ZeebeClientConfigurationProperties.class,
+  CamundaClientProperties.class
+})
 @WireMockTest(httpPort = 14682)
-public class AuthenticationConfigurationOidcTest {
+public class CredentialsProviderSelfManagedTest {
```

Comment:
💭 What do you think about adding a similar test with legacy properties ? Something like `LegacyCredentialsProviderSelfManagedTest`

Reply:
Legacy properties did not support SM, only new properties started to provide it.

Reply:
What do you mean ?

Reply:
Legacy:
Only cloud authentication was supported. No authentication via Identity.

New properties:
Both cloud and Identity authentication is supported.